src/systemctl/systemctl.c

```diff
@@ -1480,16 +1480,20 @@ static int output_unit_file_list(const UnitFileList *units, unsigned c) {
 
                 id = basename(u->path);
 
-                r = unit_file_query_preset(arg_scope, arg_root, id, &presets);
-                if (r < 0) {
-                        unit_preset_str = "n/a";
-                        on_preset_color = underline ? on_underline : ansi_normal();
-                } else if (r == 0) {
-                        unit_preset_str = "disabled";
-                        on_preset_color = underline ? ansi_highlight_red_underline() : ansi_highlight_red();
-                } else {
-                        unit_preset_str = "enabled";
-                        on_preset_color = underline ? ansi_highlight_green_underline() : ansi_highlight_green();
+                if (IN_SET(u->state, UNIT_FILE_ALIAS, UNIT_FILE_STATIC))
+                        unit_preset_str = "-"; /* Don't show preset state for aliases, it'll only confuse users. */
```

Comment:
hmm, please insert a TABLE_EMPTY cell instead. That's useful for alternative ouputs (ie. JSON), and adds implicit colorouing...

i.e. use the table_add_many() below just for the first two fields. and then add the preset field with a separate call that either adds TABLE_EMPTY or TABLE_STRING depending if we have a preset or not.